app/models/collection.rb

```diff
@@ -10,9 +10,13 @@ class Collection < ApplicationRecord
   has_many :collection_users, dependent: :destroy
   has_many :users, through: :collection_users
 
-  has_many :collection_tasks, dependent: :destroy
+  has_many :collection_tasks, lambda {
+                                order(rank: :desc).order('collection_tasks.created_at ASC')
+                              }, dependent: :destroy, inverse_of: :collection
   has_many :tasks, through: :collection_tasks
```

Comment:
I noticed that it is not necessary to sort the tasks as well.  When the tasks get queried they automatically go through the sorting of the join table. The only caveat is that you have to explicitly say which columns to sort on if there are multiple with the same name (in this case tasks and collection_tasks both have the created_at column).